tests/foreman/cli/test_activationkey.py

```diff
@@ -206,6 +210,22 @@ def test_positive_create_with_usage_limit_finite(self):
         })
         self.assertEqual(new_ak['host-limit'], u'10')
 
+    def assert_negative_create_with_usage_limit(
+            self, invalid_values, *in_error_msg):
+        """Asserts Activation key is not created and respective error msg
+        :param invalid_values: invalid max-host values
+        :param in_error_msg: strings which error msg must contain
+        """
+        for limit in invalid_values:
+            with self.subTest(limit), self.assertRaises(
+                    CLIFactoryError) as raise_ctx:
+                self._make_activation_key({u'max-hosts': limit})
+            self.assert_error_msg(
+                raise_ctx,
+                u'Failed to create ActivationKey with data:',
+                *in_error_msg
+            )
```

Comment:
Maybe i'm missing something, but it looks like assertion appears to be placed outside of subtest, isn't it?